### ISSUE TITLE
StackOverflowError in data subscriptions when RavenDB server becomes unhealthy

### ISSUE THREAD

**Issue by vijayakumarsuraj** [ravendb/jvm-client#74]
The stacktrace is essentially
```
...
at net.ravendb.client.http.RequestExecutor.send(RequestExecutor.java:890)
at net.ravendb.client.http.RequestExecutor.sendRequestToServer(RequestExecutor.java:869)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:735)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:686)
at net.ravendb.client.http.RequestExecutor.handleServerDown(RequestExecutor.java:1337)
at net.ravendb.client.http.RequestExecutor.sendRequestToServer(RequestExecutor.java:876)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:735)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:686)
at net.ravendb.client.http.RequestExecutor.handleServerDown(RequestExecutor.java:1337)
at net.ravendb.client.http.RequestExecutor.sendRequestToServer(RequestExecutor.java:876)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:735)
at net.ravendb.client.http.RequestExecutor.execute(RequestExecutor.java:686)
at net.ravendb.client.http.RequestExecutor.handleServerDown(RequestExecutor.java:1337)
at net.ravendb.client.http.RequestExecutor.sendRequestToServer(RequestExecutor.java:876)
...
```

We're using the v5.4.0 client sdk, and the server is v5.4.109.

**Comment by ml054** [comment]
@vijayakumarsuraj 

Thanks for letting us know about your issue. I've moved ticket to our official bug tracker:
https://issues.hibernatingrhinos.com/issue/RDBC-759/StackOverflowError-in-data-subscriptions-when-RavenDB-server-becomes-unhealthy

You can track issue progress there.

**Comment by vijayakumarsuraj** [comment]
Ah. Thank you!

**Comment by ml054** [comment]
@vijayakumarsuraj 

Closed with #75
